solidity/contracts/token/HypNativeMovable.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+import {HypNative} from "./HypNative.sol";
+import {MovableCollateralRouter, ValueTransferBridge} from "./libs/MovableCollateralRouter.sol";
+
+// ============ External Imports ============
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+/**
+ * @title Adds rebalancing capability to HypNative.
+ * @author Abacus Works
+ */
+contract HypNativeMovable is HypNative, MovableCollateralRouter {
+    constructor(uint256 _scale, address _mailbox) HypNative(_scale, _mailbox) {}
+
+    function _rebalance(
+        uint32 domain,
+        bytes32 recipient,
+        uint256 amount,
+        ValueTransferBridge bridge
+    ) internal override {
+        bridge.transferRemote{value: amount}({
```

Comment:
the value required by the call may be greater than amount, how should we handle that?

Reply:
The same issue exists with the ERC20 collateral contracts. If we want to send 10 usdc and the bridge charges a 10% fee, should we call `usdc.approve(bridge, 11)` or `usdc.approve(bridge, 10)` (bridge keeps 1 usdc, and we receive 9 usdc on the remote chain)? 

Seems simpler to just send the bridge an amount (whether the bridge takes it from us with `transferFrom` or we send it with msg.value), and let it take a fee from that amount.

Reply:
Updated this function to pass in `msg.value` as a fee in addition to the amount we wish to move. If the value required by the call is greater than `amount`, it will have to be covered with `msg.value`.